tests/integration/test_cloudformation.py

```diff
@@ -628,3 +689,83 @@ def check_stack():
         )
 
         self.assertEqual(len(rs['logGroups']), 0)
+
+    def test_cfn_handle_elasticsearch_domain(self):
+        stack_name = 'stack-%s' % short_uid()
+        domain_name = 'es-%s' % short_uid()
+
+        cloudformation = aws_stack.connect_to_service('cloudformation')
+        es_client = aws_stack.connect_to_service('es')
+
+        cloudformation.create_stack(
+            StackName=stack_name,
+            TemplateBody=TEST_TEMPLATE_10,
+            Parameters=[{'ParameterKey': 'DomainName', 'ParameterValue': domain_name}]
+        )
+
+        # wait for deployment to finish
```

Comment:
nitpick: could we create a small util function in this class, to avoid duplication with lines 728-733 and 757-762 (and possibly other places in this file):
```
def _await_stack_completion(self, stack_name):
    def check_stack():
        stack = get_stack_details(stack_name)
        self.assertEqual(stack['StackStatus'], 'CREATE_COMPLETE')

    retry(check_stack, retries=3, sleep=2)
```